### ISSUE TITLE
How to get currentDestination using DestinationsNavigator ?

### ISSUE THREAD

**Issue by yoobi** [raamcosta/compose-destinations#693]
Hello ! 

My current use case is something like this, navigating to another destination and removing the current one from the backstack, I don't see any currentDestination using DestinationsNavigator.

```kt
navigator.navigate(NextScreenDestination) {
    popUpTo(currentDestination) {
        inclusive = true
    }
}
```

**Comment by DimitarGR8** [comment]
Based on what I came up with as a solution, I guess it should be something like this. You should directly pass the destination.

navigator.navigate(NextScreenDestination) {
    popUpTo(CurrentScreenDestination) {
        inclusive = true
    }
}

**Comment by yoobi** [comment]
Yes that could be a solution, this composable is used by many destination, it would require to add a new parameter `currentDestination`

As DestinationsNavigator is meant to be a wrapper around NavController, it could be nice to have currentDestination in it as well

**Comment by DimitarGR8** [comment]
Previously I used to pull it off like this:     

private fun executeNavigation(
        navController: NavHostController,
        direction: Direction?,
        addToBackstack: Boolean?
    ) {
        direction?.let { dir ->
            navController.navigate(dir) {
                if (addToBackstack != true) {
                    this.popUpTo(navController.currentBackStackEntry?.destination?.route.toString()) {
                        inclusive = true
                    }
                }
            }
        }
}
    
 However, it is no longer available in the v2.

**Comment by yoobi** [comment]
I'm not using v2 sorry :/
